### DISCUSSION TITLE
Order of states in state-space model

### DISCUSSION THREAD

**Discussion by AnkitNRT** [ImperialCollegeLondon/sharpy#199] [Q&A]
I am trying to implement a custom PID controller and for that, I must know the order of the states in the matrix after using the linear assembler module. Please help me in understanding the order of X(states vector) from LinearAssembler.

**Comment by ngoiz** [comment]
@AnkitNRT,

Please see the discussion https://github.com/ImperialCollegeLondon/sharpy/discussions/156 which has a lot on this and the supporting publications on SHARPy, in particular:
Goizueta, N., Wynn, A., & Palacios, R. (2021). Parametric Krylov-based order reduction of aircraft aeroelastic models. In AIAA SciTech Forum (pp. 1–25). https://doi.org/10.2514/6.2021-1798

Thanks,
Norberto

**Reply by AnkitNRT** [reply]
I will look into the references you provided.
Thank you,
Norberto

**Reply by AnkitNRT** [reply]
Is there any reference that explains the basic modeling of vehicle in SHARPy and placement of custom control surfaces?
